spec/components/tab_navigation_component_spec.rb

```diff
@@ -25,14 +25,25 @@
 
   context 'with link for current request' do
     let(:request_path) { '/first' }
+    let(:request_method) { 'GET' }
 
     around do |example|
       Rails.application.routes.draw do
         get '(:example_param)/first' => 'application#first'
+        post '(:example_param)/first' => 'application#first_create'
         get '(:example_param)/second' => 'application#second'
+        post '(:example_param)/second' => 'application#second_create'
       end
 
-      with_request_url(request_path) { example.run }
+      with_request_url(request_path) do
+        vc_test_request.request_method = request_method
+        vc_test_request.path_parameters = Rails.application.routes.recognize_path_with_request(
+          vc_test_request,
+          request_path,
+          {},
+        )
```

Comment:
`with_request_url` doesn't provide an option to specify a request method, so I had to manually recreate some of these bits using internals of the helper. Might be worth an upstream pull request, since it seems easy enough to parameterize.

https://github.com/ViewComponent/view_component/blob/095f76f7140260402741293716d8a7472c0d9e4a/lib/view_component/test_helpers.rb#L179

Reply:
Upstream: https://github.com/ViewComponent/view_component/pull/1805